src/DependencyInjection/Compiler/GuardCompilerPass.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the BoShurikTelegramBotBundle.
+ *
+ * (c) Alexander Borisov <boshurik@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace BoShurik\TelegramBotBundle\DependencyInjection\Compiler;
+
+use BoShurik\TelegramBotBundle\DependencyInjection\BoShurikTelegramBotExtension;
+use BoShurik\TelegramBotBundle\Guard\TelegramAuthenticator;
+use BoShurik\TelegramBotBundle\Guard\TelegramLoginValidator;
+use BoShurik\TelegramBotBundle\Guard\UserLoaderInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class GuardCompilerPass implements CompilerPassInterface
```

Comment:
I think it can be places in BoShurikTelegramBotExtension.
e.g.
```php
if ($config['guard']['enabled']) {
    $loader->->load('guard.yaml');

    $definition = $container->getDefinition(TelegramAuthenticator::class);
    $definition->setArgument(3, $configs['guard_route']);
    $definition->setArgument(4, $configs['default_target_route']);
    $definition->setArgument(5, $configs['login_route'] ?? null);
}
```

Reply:
What about this? Is it impossible to do it in `BoShurikTelegramBotExtension`?

Reply:
What's the problem here?
I don't understand.

Reply:
Why did you put this code in CompilerPass not in BoShurikTelegramBotExtension (e.g. in [FrameworkBundle](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L840-L846))?

Reply:
@bigfoot90 I mean it can be simplified to something like this:
```yaml
#guard.yaml
services:
    BoShurik\TelegramBotBundle\Guard\TelegramLoginValidator:
        arguments:
            - '%boshurik_telegram_bot.api.token%'

    BoShurik\TelegramBotBundle\Guard\TelegramAuthenticator:
        arguments:
            - '@BoShurik\TelegramBotBundle\Guard\TelegramLoginValidator'
            - '@BoShurik\TelegramBotBundle\Guard\UserLoaderInterface'
            - '@Symfony\Component\Routing\Generator\UrlGeneratorInterface'
            - '%boshurik_telegram_bot.gurad.guard_route%'
            - '%boshurik_telegram_bot.gurad.default_target_route%'
            - '%boshurik_telegram_bot.gurad.login_route%'
```
```php
// BoShurikTelegramBotExtension::load

if ($config['guard']['enabled']) {
    $loader->load('guard.yaml');

    $container->setParameter('boshurik_telegram_bot.gurad.guard_route', $config['gurad']['guard_route']);
    $container->setParameter('boshurik_telegram_bot.gurad.default_target_route', $config['gurad']['guard_route']);
    $container->setParameter('boshurik_telegram_bot.gurad.login_route', $config['gurad']['guard_route']);
}
```

Reply:
@BoShurik Updated! Ready for review.